redbot/cogs/mod/names.py

```diff
@@ -69,6 +69,96 @@ class ModInfo(MixinMeta):
+            return None, discord.ActivityType.custom
+        a = a[0]
+        c_status = None
+        if not a.name and not a.emoji:
+            return None, discord.ActivityType.custom
+        elif a.name and a.emoji:
+            c_status = _("Custom: {emoji} {name}").format(emoji=a.emoji, name=a.name)
+        elif a.emoji:
+            c_status = _("Custom: {emoji}").format(emoji=a.emoji)
+        elif a.name:
+            c_status = _("Custom: {name}").format(name=a.name)
+        return c_status, discord.ActivityType.custom
+
+    def handle_playing(self, user):
+        p_acts = [c for c in user.activities if c.type == discord.ActivityType.playing]
+        p_act = p_acts[0] if p_acts else None
+        act = (
+            _("Playing: {name}").format(name=p_act.name if p_act and p_act.name else None)
+            if p_act and hasattr(p_act, "name")
+            else p_act.name
+            if p_act and p_act.name
+            else None
+        )
+        return act, discord.ActivityType.playing
+
+    def handle_streaming(self, user):
+        s_acts = [c for c in user.activities if c.type == discord.ActivityType.streaming]
+        s_act = s_acts[0] if s_acts else None
+        act = (
+            _("Streaming on {platform}: [{name}{sep}{game}]({url})").format(
+                platform=s_act.platform,
+                name=discord.utils.escape_markdown(s_act.name),
+                sep=" | " if s_act.game else "",
+                game=discord.utils.escape_markdown(s_act.game) if s_act.game else "",
+                url=s_act.url,
+            )
+            if s_act and s_act.name and hasattr(s_act, "game")
+            else discord.utils.escape_markdown(s_act.name)
+            if s_act and s_act.name
+            else None
+        )
+        return act, discord.ActivityType.streaming
+
+    def handle_listening(self, user):
+        l_acts = [c for c in user.activities if c.type == discord.ActivityType.listening]
+        l_act = l_acts[0] if l_acts else None
+        act = (
+            _("Listening: [{title}{sep}{artist}]({url})").format(
+                title=l_act.title,
+                sep=" | " if l_act.artists[0] else "",
+                artist=discord.utils.escape_markdown(l_act.artist) if l_act.artist else "",
+                url=f"https://open.spotify.com/track/{l_act.track_id}",
+            )
+            if l_act and hasattr(l_act, "title")
+            else l_act.name
+            if l_act and l_act.name
+            else None
+        )
+        return act, discord.ActivityType.listening
+
+    def handle_watching(self, user):
+        w_acts = [c for c in user.activities if c.type == discord.ActivityType.watching]
+        w_act = w_acts[0] if w_acts else None
+        act = (
+            _("Watching: {name}").format(name=w_act.name if w_act else None)
+            if w_act and hasattr(w_act, "name")
+            else w_act.name
+            if w_act and w_act.name
+            else None
+        )
```

Comment:
This seems overcomplicated - activities always have a `name` attribute. Unless there's more to it than I see, you should be able to just simplify it to:
```suggestion
        if not w_acts:
            return None, discord.ActivityType.watching
        w_act = w_acts[0]
        act = _("Watching: {name}").format(name=w_act.name)
```